hnn_core/network.py

```diff
@@ -432,31 +459,38 @@ def add_poisson_drive(self, name, *, tstart=0, tstop=None, rate_constant,
         if tstop is None:
             tstop = sim_end_time
 
-        if not self._legacy_mode:
-            _check_drive_parameter_values('Poisson', tstart=tstart,
-                                          tstop=tstop,
-                                          sim_end_time=sim_end_time)
-            target_populations = _get_target_populations(weights_ampa,
-                                                         weights_nmda)[0]
-            _check_poisson_rates(rate_constant, target_populations,
-                                 self.cell_types.keys())
+        _check_drive_parameter_values('Poisson', tstart=tstart,
+                                      tstop=tstop,
+                                      sim_end_time=sim_end_time)
+        target_populations = _get_target_populations(weights_ampa,
+                                                     weights_nmda)[0]
+        _check_poisson_rates(rate_constant, target_populations,
+                             self.cell_types.keys())
+        if isinstance(rate_constant, dict):
+            if n_drive_cells != 'n_cells':
+                raise ValueError(f"For Poisson rate constants (got "
+                                 f"{rate_constant}) that are cell type "
+                                 f"specific, 'n_drive_cells' must be set "
+                                 f"to 'n_cells' so that the drive has 1-to-1"
+                                 f"connectivity")
```

Comment:
Rephrase slightly to be more intelligible?

"Drives specific to cell types is only possible with cell_specific=True and n_drive_cells = 'n_cells'. Got cell_specific={cell_specific} and n_drive_cells={n_drive_cells}"